sdks/python/apache_beam/io/textio.py

```diff
@@ -117,7 +123,7 @@ def display_data(self):
         label='Coder')
     return parent_dd
 
-  def read_records(self, file_name, range_tracker):
+  def read_records(self, file_name, range_tracker, process_header_fn=None):
```

Comment:
<!--new_thread; commit:7967f9db74c71484c9b94f42bf5c5219cbb6cb85; resolved:0-->
Please don't update read_records() method here which is a part of the FileBasedSource interface. How about making the new TextSource.__init__() parameter a function 'header_processor' to which header lines are passed ? You can do matching within the this processor. (You'll have to instantiate TextSource in VcfSource.read_records()).

Reply:
Sounds good! I moved the header_processor function to be in the init function, but we still need the predicate_matcher to detect which lines are part of the header in the `_process_header` function.